### DISCUSSION TITLE
Where is the `@expo/xcpretty` source code?

### DISCUSSION THREAD

**Discussion by ngregory-rbi** [expo/expo#18646] [Archived: Questions] [answered]
Hello!

I'm trying to track down a bug in my application's build process and I see a log I have narrowed down to `@expo/xcpretty`. My `node_modules` includes the compiled `.js` files for the package, and I was wondering where the source code is. The bug itself I can narrow down, it would just be easier for me to do that if I had the source code for `@expo/xcpretty` to trace the log message back. I did some looking through this repo and `expo-cli` and was unable to find it. Is it not available?

**Comment by keith-kurak** [comment, marked as answer]
Hi, @ngregory-rbi !

This is a private package, so the source (other than what's in the node_modules build folder) is not available. However, I wonder if whatever error message you're seeing is actually caused by something else, as this is simply a package for formatting the display of Xcode logs. We may be able to offer some suggestions if you post the actual error message.

**Reply by ngregory-rbi** [reply]
Hey Keith! I actually did end up solving the issue. The reason I wanted to look at the source code is that the error message itself was quite vague and I wanted to trace it back through to see what could have generated it.

I will ask though, why is it a private package? I don't see anything about it that would preclude it from being published.

**Reply by keith-kurak** [reply]
Glad to hear it's working! @expo/xcpretty is currently private because it's part of a toolchain that deals with API’s that are in flux, and thus not well-positioned at this time for external contribution.

**Reply by ngregory-rbi** [reply]
Thanks for the info :)

**Reply by EvanBacon** [reply]
For future reference, `@expo/xcpretty` has xcodebuild log data piped to it, then formats by matching fragile regexes against the log output. We store the log data in the project's `.expo/xcodebuild.log` file so you should be able to see the unfiltered error messages there. You can also run the log output through a different logger by doing something like `cat .expo/xcodebuild.log | xcpretty`.
